app/scripts/directives/serviceIntegration.js

```diff
@@ -0,0 +1,241 @@
+            _.set(iterateBinding, "metadata.annotations.consumer", ctrl.consumerService.metadata.labels.serviceName);
+            _.set(iterateBinding, "metadata.annotations.provider", ctrl.integration);
+            DataService.update(APIService.getPreferredVersion("servicebindings"), iterateBinding.metadata.name, iterateBinding, context)
+          }
+        });
+      });
+
+
+      DataService.create(version, null, podPreset, context)
+      .then(function() {
+        return DataService.get(APIService.getPreferredVersion('deployments'), _.get(ctrl.consumerService, 'metadata.labels.serviceName'), context, {
+          errorNotification: false
+        });
+      })
+      .then(function(deployment) {
+        deployment.spec.template.metadata.labels[_.get(ctrl.serviceInstance, 'metadata.labels.serviceName')] = "enabled";
+        return DataService.update(APIService.getPreferredVersion('deployments'), _.get(ctrl, 'consumerService.metadata.labels.serviceName'), deployment, context);
+      })
+      .then(function() {
+        NotificationsService.addNotification({
+          type: "success",
+          message: "Binding has been set in " + _.get(ctrl, 'consumerService.metadata.labels.serviceName') + " Redeployed " + _.get(ctrl, 'consumerService.metadata.labels.serviceName')
+        });
+      })
+      .catch(function(err) {
+        NotificationsService.addNotification({
+          type: "error",
+          message: "Failed to integrate service binding",
+          details: err.data.message
+        });
+      });
+    };
+
+    ctrl.getState = function(){
+      if(ctrl.podPreset && !ctrl.binding) {
+        return 2;
+      }
+      if(ctrl.podPreset && ctrl.binding) {
+        return 1;
+      }
+      if(ctrl.binding) {
+        var state=1;
+        _.each(ctrl.binding.status.conditions, function(condition){
+          if(condition.type === "Ready" && condition.status !== "True"){
+            state=2;
+          }
+        });
+        return state;
+      }
+      if (ctrl.serviceInstance){
+        return 0;
+      }
+      return -1;
+    };
+
+    //called in callback from succesful delete-link for binding
+    ctrl.deletePodPreset = function(){
+      var context = {namespace: ctrl.consumerService.metadata.namespace};
+      var deleteOptions = {propagationPolicy: null};
+      DataService.delete(podPresetVersion, ctrl.podPreset.metadata.name, context, deleteOptions)
+      .then(function(){
+        return DataService.get(APIService.getPreferredVersion('deployments'), _.get(ctrl, 'consumerService.metadata.labels.serviceName'), context)
+      })
+      .then(function(deployment) {
+        //if there is a label in the deployment named after the integrated service, 
+        //delete the label and update the deployment to trigger a redeploy
+        if(ctrl.integrationData.name in deployment.spec.template.metadata.labels) {
+          delete(deployment.spec.template.metadata.labels[ctrl.integrationData.name]);
```

Comment:
If `deployment.spec.template.metadata.labels` exists we can just delete as it won't error.

delete doesn't need the brackets either